code/modules/research/xenobiology/crossbreeding/destabilized.dm

```diff
@@ -0,0 +1,322 @@
+/*
+Destabilized extracts:
+	Throwable extracts that apply an effect on hit.
+*/
+
+/obj/item/slimecross/destabilized
+	name = "destabilized extract"
+	desc = "It seems ready to burst at any moment."
+	effect = "destabilized"
+	icon_state = "destabilized"
+	amountToCreate = 5
+
+/obj/item/slimecross/destabilized/throw_impact(atom/A, var/datum/thrownthing/T)
+	if(!..()) //For those tactical catches
+		if(hitEffect(A, T.thrower))
+			playsound(get_turf(A), 'sound/misc/splort.ogg', get_volume_by_throwforce_and_or_w_class(), 1, -1)
+			qdel(src)
+
+/obj/item/slimecross/destabilized/proc/hitEffect(atom/A, mob/thrower)
```

Comment:
Fairly certain thrower isn't always a mob.

Doesn't look like you accounted for that.

Reply:
Is the other case chutes?

Reply:
Disposal chutes, mass drivers and some other things assert themselves as *throwers*. Not sure how I even feel about that happening but its a thing.

Reply:
@nemvar I just checked, and it seems the thrownthing datum that I'm referencing initialized it's thrower as a mob, so I'm going to assume it handles that on it's end?

Reply:
It looks like the reason the extracts work with the pneumatic cannon specifically is because it acts as if the user is throwing it. For all other cases where a thrownthing is not initialized, the extracts just wont react, which is fine since most require a specific thrower for their functionality.